src/operators.jl

```diff
@@ -858,3 +855,28 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, v::Vector{U})  where U <: Integer
+    v0 = minimum(v)
+    vs = Set(v)
+    filter!(x -> x != v0,vs)
```

Comment:
You don't need to do this. You know `v0` is in `vs` exactly once, so you can just do `delete!(vs, v0)` without having to traverse the entire set via a `filter`.